### PR TITLE
Upgrade java dependencies

### PR DESCRIPTION
**Pull request by Rappahannock**
* Upgrade java dependencies - spring boot 2.6.8 + other libs
* Upgrade to java 11 (latest test libs upgrade requires it)

### CODE REVIEW

File: src/test/java/org/redischool/attendance/FeatureTestForSecurity.java [hunk 1]
```diff
@@ -17,98 +18,99 @@
...
+@SpringBootTest(
+        webEnvironment = DEFINED_PORT,
+        properties = {
+                "credentials.username=foo",
+                "credentials.password=bar"
+        }
+)
+@RunWith(SpringJUnit4ClassRunner.class)
+@ActiveProfiles("cloud")
+@Ignore("Currently failing, investigation in progress")
```
Comment (jacekbilski) [inline src/test/java/org/redischool/attendance/FeatureTestForSecurity.java]: Those tests are not yet fixed and I'll have slight problems approving this pull request.

File: build.gradle [hunk 2]
```diff
@@ -31,9 +31,9 @@ dependencies {
     compile("org.springframework.boot:spring-boot-starter-web:${springBootVersion}")
     compile("org.springframework.boot:spring-boot-starter-thymeleaf:${springBootVersion}")
     compile('org.springframework.boot:spring-boot-starter-security')
-    compile('org.springframework.security.oauth.boot:spring-security-oauth2-autoconfigure:2.0.1.RELEASE')
-    compile('org.springframework.security.oauth:spring-security-oauth2:2.2.0.RELEASE')
-    compile('com.google.apis:google-api-services-sheets:v4-rev516-1.23.0')
+    compile("org.springframework.security.oauth.boot:spring-security-oauth2-autoconfigure:${springBootVersion}")
```
Comment (erikpetzold) [inline build.gradle:34]: not sure if this always the same version like spring boot version

but as far as I understand this dependency will be removed in the next step completely, so does not matter?

**Review by erikpetzold — COMMENTED** [review]
Looks good to me, not sure about the tests - will they be fixed here or in the branch of Jacek?

File: src/test/java/org/redischool/attendance/FeatureTestForSecurity.java (hunk 1 above)
Comment (jacekbilski) [reply src/test/java/org/redischool/attendance/FeatureTestForSecurity.java]: Once #17 is merged, this problem will be solved.

File: build.gradle (hunk 2 above)
Comment (Rappahannock) [reply build.gradle:34]: I upgraded in several steps (by gradually updating spring boot version) and checked how version of this lib correlates with used spring boot version, it was always 1-to-1.

File: src/test/java/org/redischool/attendance/FeatureTestForSecurity.java (hunk 1 above)
Comment (Rappahannock) [reply src/test/java/org/redischool/attendance/FeatureTestForSecurity.java]: I rebased on master, squashed commits + did some extra minor improvements to the tests (that my IDE showed as problems). I think it is good to go now.
I also smoke-tested the app by running it locally.

File: src/test/java/org/redischool/attendance/details/CourseDetailsRepositoryTest.java [hunk 3]
```diff
@@ -125,8 +125,8 @@ public class CourseDetailsRepositoryTest {
 
             describe("update course", () -> {
                 it("doesn't do anything with empty update data", () -> {
-                    courseDetailsRepository.updateAttendance("1", "", new HashMap<String, String>());
-                    verifyZeroInteractions(googleSheetsApi);
+                    courseDetailsRepository.updateAttendance("1", "", new HashMap<>());
+                    verifyNoMoreInteractions(googleSheetsApi);
```
Comment (jacekbilski) [inline src/test/java/org/redischool/attendance/details/CourseDetailsRepositoryTest.java]: That's not true, it's changing the meaning of the assertion. Before we were making sure we didn't interact with GoogleSheetsApi at all. Now we're theoretically allowing some interactions and then making sure nothing else has happened. What we want are zero interactions, please revert.

Comment (Rappahannock) [reply src/test/java/org/redischool/attendance/details/CourseDetailsRepositoryTest.java]: That was a mistake when I replaced the deprecated verifyZeroInteractions, I updated it now to verifyNoInteractions.

**Review by jacekbilski — APPROVED** [review]
If all the tests are green you're good to go.

**Review by erikpetzold — APPROVED** [review]

**Pull request merged**